Backpack-SwiftUI/Select/Classes/BPKSelect.swift

```diff
@@ -0,0 +1,170 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018-2022 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+
+struct CustomPickerStyle: ViewModifier {
+    var labelText: String
+    var textColor: BPKColor
+    
+    func body(content: Content) -> some View {
+        Menu {
+            content
+        } label: {
+            HStack {
+                if let labelText = labelText {
+                    BPKText(
+                        labelText
+                    )
+                    .foregroundColor(textColor)
+                    Spacer()
+                    Image(systemName: "triangle.fill")
+                        .resizable()
+                        .foregroundColor(.textPrimaryColor)
+                        .frame(width: 8, height: 5)
+                        .rotationEffect(.degrees(180))
+                }
+            }
+            .frame(height: .lg)
+        }
+        .padding(.md)
+    }
+}
+
+extension View {
+    func customPickerStyle(labelText: String,
+                           textColor: BPKColor) -> some View {
+        self.modifier(
+            CustomPickerStyle(
+                labelText: labelText,
+                textColor: textColor
+            )
+        )
+    }
+}
+
+public struct BPKSelect: View {
+    
+    @SwiftUI.State private var selectedIndex: Int
+
+    private let options: [String]
+    private let title: String
+    private var state: State = .default
+    private let onSelectionChange: (Int) -> Void
+    
+    private var indexIdentifiableOptions: [IdentifiableString] {
+        options.enumerated().map { (index, element) in
+            IdentifiableString(string: element, id: index)
+        }
+    }
+    
+    public init(
+        placeholder: String,
+        options: [String],
+        selectedIndex: Int,
+        onSelectionChange: @escaping (Int) -> Void
+    ) {
+        self.options = options
+        self.selectedIndex = selectedIndex
+        self.title = placeholder
+        self.onSelectionChange = onSelectionChange
+    }
+    
+    public var body: some View {
+        VStack {
+            
+            Picker(title, selection: $selectedIndex) {
+                ForEach(indexIdentifiableOptions) { t in
+                    Text(t.string)
+                }
+            }
+            .customPickerStyle(
+                labelText: selectedIndex < options.count ? options[selectedIndex] : "",
```

Comment:
I think we may benefit from using the `.pickerStyle(.menu)` api to create our picker style, rather than a generic view extension.

Have you given this a try?

Right now doing this bit of logic here is making the Placeholder to never be shown, as there's always an empty string

Reply:
.pickerStyle(.menu) looks very different to the designs, this was how I was able to make it look more like the sigma. Happy to discuss options, but I went through the simple approach 1st, and unless we're able to push back a lot on the designs I don't think .pickerStyle(.menu) is going to suffice.

Reply:
With the following code:

```
Picker(title, selection: $selectedIndex) {
                ForEach(indexIdentifiableOptions) { t in
                    Text(t.string)
                }
            }.pickerStyle(.menu)
```

It looks like this:

![Screenshot 2023-07-20 at 14 59 18](https://github.com/Skyscanner/backpack-ios/assets/391610/eb1c415e-e318-4765-9ca5-78754f45395d)

Please note:
1. The placeholder is still not shown
2. The style looks completely different to Figma


![Screenshot 2023-07-20 at 15 00 56](https://github.com/Skyscanner/backpack-ios/assets/391610/b2d8269c-dbfa-407d-9448-ba54d7df2662)

Reply:
Ahh!! all good around this then!!

Reply:
changed in: https://github.com/Skyscanner/backpack-ios/pull/1708/commits/e869ab618cf4f5b4bd63c6bcc874610a21b8d83a